### PR TITLE
[SYCL] Bump libclc to OpenCL 3.0

### PR DESCRIPTION
**Pull request by hdelan**
This is needed to enable the generic AS in libclc which is useful for atomic ops

### CODE REVIEW

**Comment by hdelan** [comment]
@bader Can I get this reviewed. I have an incoming PR to add extra HIP atomics that depends on this PR. Thanks

**Comment by bader** [comment]
@hdelan, please, ping the code owners - @intel/llvm-reviewers-cuda team.

**Comment by hdelan** [comment]
Aha sorry @bader. Ping @intel/llvm-reviewers-cuda

File: libclc/CMakeLists.txt [hunk 1]
```diff
@@ -378,6 +375,18 @@ foreach( t ${LIBCLC_TARGETS_TO_BUILD} )
                 # have to be explicity declared in the soruce.
                 list( APPEND flags -Xclang -fdeclare-spirv-builtins)
 
+                # OpenCL 3.0 extensions
+                list( APPEND flags -cl-std=CL3.0 "SHELL:-Xclang" )
```
Comment (npmiller) [inline libclc/CMakeLists.txt:379]: Should this be:
```suggestion
                list( APPEND flags "SHELL:-Xclang -cl-std=CL3.0" )
```

Comment (npmiller) [reply libclc/CMakeLists.txt:379]: But it was there previously for `-cl-std=CL2.0`

File: libclc/CMakeLists.txt [hunk 2]
```diff
@@ -378,6 +375,18 @@ foreach( t ${LIBCLC_TARGETS_TO_BUILD} )
...
+                string(CONCAT CL_3_0_EXTENSIONS
+                  "-cl-ext="
+                  "+cl_khr_fp16,"
```
[the remark this answers is not in the document]
Comment (hdelan) [reply libclc/CMakeLists.txt]: Change made

**Review by npmiller — APPROVED** [review]

**Pull request merged**